### PR TITLE
feat: refactor arrow to typescript #175

### PR DESCRIPTION
**Pull request by cboyce183**
Arrow component, with added types

### CODE REVIEW

File: src/Arrow/index.tsx [hunk 1]
```diff
@@ -13,10 +13,11 @@ const transformMap = {
 	left: 'translateX(-50%) rotate(45deg)',
 };
 
-function useArrowStyles(primaryPlacement, arrowSize) {
+function useArrowStyles(primaryPlacement: string, arrowSize: Number) {
 	const arrowStyles = useMemo(
 		() => ({
-			position: 'absolute',
+			// casting field to itself to enforce type as valid
+			position: 'absolute' as 'absolute',
```
Comment (diondiondion) [inline src/Arrow/index.tsx:20]: Why's this necessary? Is there something special about the position property or the absolute value that necessitates this? If so, that should be added to the comment.

File: src/Arrow/index.tsx [hunk 2]
```diff
@@ -78,7 +79,7 @@ Arrow.propTypes = {
 	 * Controls how far the arrow is placed from the container's edge when
 	 * the placement contains a '-start' or '-end' suffix.
 	 */
-	distanceFromEdge: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+	distanceFromEdge: PropTypes.number,
```
Comment (diondiondion) [inline src/Arrow/index.tsx:82]: TypeScript types should replace PropTypes, can you convert the PropTypes definition to a type interface for the `Arrow` component?

You could check out the [`ThemeSection`](https://github.com/5app/base5-ui/blob/master/src/ThemeSection/index.tsx) component for an example.

File: src/Arrow/index.tsx [hunk 3]
```diff
@@ -37,7 +38,7 @@ function useArrowStyles(primaryPlacement, arrowSize) {
 	return arrowStyles;
 }
 
-const Arrow = forwardRef((props, ref) => {
+const Arrow = forwardRef((props: {placement: string, size: number, distanceFromEdge: number, style}, ref: React.RefObject<HTMLElement>) => {
```
Comment (diondiondion) [inline src/Arrow/index.tsx:41]: The `placement` prop can be defined more granularly based on the `PLACEMENTS` constant – or maybe you can even import that type from popperJS. Should result in a nice auto-complete for this prop, as only a limited number of strings are actually allowed as values.

**Review by diondiondion — COMMENTED** [review]
Hi Charlie, how are things? :-)

Thanks for making a start working on this. I think it'd be nice if we could phase out PropTypes as we introduce TypeScript. And should the `getArrowPosition` function be converted to TS, too? (I haven't looked at it just now, just wondering if that would make things more consistent or robust)

File: src/Arrow/index.tsx (hunk 1 above)
Comment (cboyce183) [reply src/Arrow/index.tsx:20]: Yes. It doesn't recognise the unmodified string as a valid option defined on the position interface, it recognises it simply as a string, but casting it to itself fixes this. I am not well enough versed on TS yet to explain why.

**Review by diondiondion — APPROVED** [review]
LGTM, thanks Charlie. Just some comments.

**Pull request merged**